src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -385,7 +385,14 @@ private void AddCredentialToCache(Uri serverUri, ulong authAvail, NetworkCredent
                     if ((authAvail & s_authSchemePriorityOrder[i]) != 0)
                     {
                         Debug.Assert(_credentialCache != null, "Expected non-null credential cache");
-                        _credentialCache.Add(serverUri, s_authenticationSchemes[i], nc);
+                        try
+                        {
+                            _credentialCache.Add(serverUri, s_authenticationSchemes[i], nc);
+                        }
+                        catch(ArgumentException)
+                        {
+                            //Ignore the case of key already present
```

Comment:
How common is this case?  And is it possible it could have previously been added with a different authentication scheme?